### ISSUE TITLE
Energy broken on 2021.12

### ISSUE THREAD

**Issue by pluskal** [weltenwort/home-assistant-rct-power-integration#170]
## Version of the custom_component
v0.8.2

## Configuration
Integration added in Integration UI. In 2021.11 worked as expected.

## Describe the bug
After an update to the 2021.12, the energy monitor dashboard stopped working. The energy settings show an unexpected device class.

![image](https://user-images.githubusercontent.com/3617487/145942030-5e042753-db0d-4f42-a2ca-b1a73bff3970.png)

## Debug log

This doesn't seem relevant but relates to the RCT component.
```text
Logger: homeassistant.components.sensor.recorder
Source: components/sensor/recorder.py:484
Integration: Sensor (documentation, issues)
First occurred: December 13, 2021, 8:05:10 PM (32 occurrences)
Last logged: December 13, 2021, 8:05:10 PM

The unit of sensor.rct_power_storage_generator_b_energy_production_day (Wh) does not match the unit of already compiled statistics (kWh). Generation of long term statistics will be suppressed unless the unit changes back to kWh
The unit of sensor.rct_power_storage_generator_b_energy_production_month (Wh) does not match the unit of already compiled statistics (kWh). Generation of long term statistics will be suppressed unless the unit changes back to kWh
The unit of sensor.rct_power_storage_generator_b_energy_production_year (Wh) does not match the unit of already compiled statistics (kWh). Generation of long term statistics will be suppressed unless the unit changes back to kWh
The unit of sensor.rct_power_storage_generator_b_energy_production_total (Wh) does not match the unit of already compiled statistics (kWh). Generation of long term statistics will be suppressed unless the unit changes back to kWh
The unit of sensor.rct_power_storage_all_generators_energy_production_total (Wh) does not match the unit of already compiled statistics (kWh). Generation of long term statistics will be suppressed unless the unit changes back to kWh
```

**Comment by weltenwort** [comment]
Thanks for reporting, I'll test with the new version.

I wish the HA team working on the energy dashboard would document the requirements for sensor to be usable on it - and the changes to those requirements.

**Comment by Benjamin410** [comment]
Could it be related to this change (was hint regarding device classes in release notes) https://github.com/home-assistant/core/pull/60654?

**Comment by weltenwort** [comment]
Possible, but the PR suggests the old constants are still there. So I'd merely expect a deprecation notice. But it's a good place to start the investigation, thanks.

**Comment by weltenwort** [comment]
Could you help me out by pasting the "Attributes" of one of the entities that the Energy dashboard complains about?

![image](https://user-images.githubusercontent.com/973741/146250062-295d552c-b7c0-4b95-88fd-62396be66cf8.png)

**Comment by Benjamin410** [comment]
![Screenshot_20211215-204701.png](https://user-images.githubusercontent.com/10449981/146254928-21d6cae4-d24b-46c6-aa78-275c57eceff1.png)

**Comment by Benjamin410** [comment]
![image](https://user-images.githubusercontent.com/10449981/146258748-d6419d5f-2303-4962-9718-6666f125d30e.png)

**Comment by weltenwort** [comment]
Thanks! It indeed seem to be missing the `device_class` completely.

**Comment by weltenwort** [comment]
I've [released 0.9.0](https://github.com/weltenwort/home-assistant-rct-power-integration/releases/tag/v0.9.0) with a fix. I'd appreciate any feedback you can give me about the efficacy of the fix. I hope the unit confusion in the statistics recorder resolves itself or is at least easily fixed using the "Fix it" feature of the "Statistics" table in the "Developer Tools". Again, any feedback wound be appreciated.

For anyone interested, the cause for the missing `device_class` was not the move from constants to an `Enum`, but the fact that the `SensorEntity` now overrides the base `Entity`'s `device_class` property. That caused my derived `RctPowerEntity` subclass's `device_class` to be ignored.

**Comment by pluskal** [comment]
Thanks for the fix! I have tested the `0.9.0`, and it fixed the problem.

**Comment by Benjamin410** [comment]
Thanks also from my side the fix works and it recovered the old data.
